### ISSUE TITLE
Building later version of GDAL using emscripten

### ISSUE THREAD

**Issue by sghofrany** [ddohler/gdal-js#51]
I was wondering if you could document your process of building GDAL using emscripten, it would be greatly appreciated.

**Comment by ddohler** [comment]
I'd be happy to describe what I can, but could you tell me more about what you're trying to do? I see from the issue title that you might be trying to build a later version of GDAL with emscripten; if that's true, then it's important to note that GDAL >3.0 switches to a build system based on `cmake`, which works completely differently from the build system that this project currently uses. I've put some effort into compiling GDAL 3 for use in this project, but haven't finished yet.

**Comment by sghofrany** [comment]
Yes, my goal was to use cmake with GDAL >3.0. Do you see this as being an issue? I've noticed that emscripten has a wrapper for cmake `emcmake cmake .` for example.

**Comment by ddohler** [comment]
No, it is definitely possible, I know that at least one other person has already gotten GDAL 3 to compile with Emscripten: https://github.com/bugra9/gdal3.js . The goals of that project and this one are somewhat different so I haven't been able to copy what they did, but I am checking my work against theirs as I go because there are a lot of similarities.

**Comment by sghofrany** [comment]
If I wanted to do something like this using cmake, would I have to compile both PROJ and GDAL into emscripten or just GDAL? I'm still a little confused about what parts of my project need to be converted.

**Comment by ddohler** [comment]
> If I wanted to do something like this using cmake, would I have to compile both PROJ and GDAL into emscripten or just GDAL?

You'll need to compile all dependencies of your project with emscripten, including PROJ and GDAL; see here: https://github.com/bugra9/gdal3.js/blob/master/Makefile#L191

**Comment by sghofrany** [comment]
Ahh interesting. I use the default GDAL and PROJ builds for work, will building these targets using emscripten cause any issues to already installed targets?

**Comment by ddohler** [comment]
> Ahh interesting. I use the default GDAL and PROJ builds for work, will building these targets using emscripten cause any issues to already installed targets?

If I'm understanding your question correctly, I don't think so -- building with emscripten will simply generate new copies of the compiled library within the project directory; it shouldn't affect any copies you have installed anywhere else on your system.

**Comment by sghofrany** [comment]
Great, thanks for your help and insight. If it's okay with you I'd like to keep this question open for other questions later on.
